homeassistant/components/lastfm/sensor.py

```diff
@@ -35,95 +35,54 @@
-
-    add_entities(entities, True)
-
-
-class LastfmSensor(SensorEntity):
+class LastFmSensor(SensorEntity):
     """A class for the Last.fm account."""
 
     _attr_attribution = "Data provided by Last.fm"
     _attr_icon = "mdi:radio-fm"
 
-    def __init__(self, user, lastfm_api):
+    def __init__(self, user: str, lastfm_api: LastFMNetwork) -> None:
         """Initialize the sensor."""
-        self._unique_id = hashlib.sha256(user.encode("utf-8")).hexdigest()
-        self._user = lastfm_api.get_user(user)
-        self._name = user
-        self._lastfm = lastfm_api
-        self._state = "Not Scrobbling"
-        self._playcount = None
-        self._lastplayed = None
-        self._topplayed = None
-        self._cover = None
-
-    @property
-    def unique_id(self):
-        """Return the unique ID of the sensor."""
-        return self._unique_id
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def native_value(self):
-        """Return the state of the sensor."""
-        return self._state
+        self._attr_unique_id = hashlib.sha256(user.encode("utf-8")).hexdigest()
+        self._attr_name = user
+        try:
+            self._user = lastfm_api.get_user(user)
+        except WSError as exc:
+            LOGGER.error(exc)
+            self._attr_available = False
 
     def update(self) -> None:
         """Update device state."""
-        self._cover = self._user.get_image()
-        self._playcount = self._user.get_playcount()
-
-        if recent_tracks := self._user.get_recent_tracks(limit=2):
-            last = recent_tracks[0]
-            self._lastplayed = f"{last.track.artist} - {last.track.title}"
-
+        self._attr_entity_picture = self._user.get_image(SIZE_SMALL)
+        if self._user.get_now_playing() is not None:
+            self._attr_native_value = format_track(self._user.get_now_playing())
+        else:
+            self._attr_native_value = STATE_NOT_SCROBBLING
+        top_played = None
         if top_tracks := self._user.get_top_tracks(limit=1):
-            top = str(top_tracks[0])
-            if (toptitle := re.search("', '(.+?)',", top)) and (
-                topartist := re.search("'(.+?)',", top)
-            ):
-                self._topplayed = f"{topartist.group(1)} - {toptitle.group(1)}"
-
-        if (now_playing := self._user.get_now_playing()) is None:
-            self._state = STATE_NOT_SCROBBLING
-            return
-
-        self._state = f"{now_playing.artist} - {now_playing.title}"
-
-    @property
-    def extra_state_attributes(self):
-        """Return the state attributes."""
-        return {
-            ATTR_LAST_PLAYED: self._lastplayed,
-            ATTR_PLAY_COUNT: self._playcount,
-            ATTR_TOP_PLAYED: self._topplayed,
+            top_played = format_track(top_tracks[0].item)
+        last_played = format_track(self._user.get_recent_tracks(limit=1)[0].track)
```

Comment:
Are you sure that `get_recent_tracks` will always return a value?
Should it not be handled instead like `top_tracks`, using a walrus (`:=`) to ensure that some data is available?